pages/api/user/[id].js

```diff
@@ -76,6 +76,18 @@ export default async function handler(req, res) {
         return res.status(500).json({ error: error.message });
       }
     }
+
+    if (req.method === "PUT") {
+      const { id } = req.body;
+      try {
+        await Method.findByIdAndUpdate(id, {
+          $set: req.body,
+        });
```

Comment:
this is a bit hard to read since everything is implicit. you could add some variable names which would act a bit like comments in your code such that it is easier to understand.